addons/mail/static/src/models/composer_view.js

```diff
@@ -619,6 +619,16 @@ registerModel({
             }
             return clear();
         },
+        /**
+         * @private
+         * @returns {boolean|FieldCommand}
+         */
+        _computeHasThreadName() {
+            if (this.threadView && this.threadView.hasComposerThreadName !== undefined) {
```

Comment:
`this.threadView.hasComposerThreadName` default to `false` therefore it should never be `undefined`.
I don't understand why you test `undefined` specifically: 
```javascript
if (this.threadView && this.threadView.hasComposerThreadName) {
```

Reply:
Good catch, this is unnecessary. Will update this PR.

Note that this PR required turning an optional prop to field, namely `ThreadViewer/threadView_hasComposerThreadTyping`. Since it's an optional boolean, we had to deal with `undefined`.

`ThreadView/hasComposerThreadName` is strictly a boolean, but if you look at its compute method, you see `ThreadViewer/threadView_hasComposerThreadTyping` is considered only when defined

Reply:
With change, this only becomes a guard on `this.threadView`, as `ThreadView/hasComposerThreadName` is always set as you said.